src/main/java/org/fogbowcloud/manager/core/plugins/identity/openstack/KeystoneV3IdentityPlugin.java

```diff
@@ -0,0 +1,268 @@
+	}
+
+	public KeystoneV3IdentityPlugin(Properties properties) {
+		this.keystoneUrl = properties.getProperty(IDENTITY_URL) == null ? properties.getProperty(AUTH_URL)
+				: properties.getProperty(IDENTITY_URL);
+		this.v3TokensEndpoint = keystoneUrl + V3_TOKENS_ENDPOINT_PATH;
+	}
+
+	@Override
+	public Token createToken(Map<String, String> credentials) {
+
+		LOGGER.debug("Creating new Token");
+
+		JSONObject json;
+		try {
+			json = mountJson(credentials);
+		} catch (JSONException e) {
+			LOGGER.error("Could not mount JSON while creating token.", e);
+
+			Integer statusResponse = HttpStatus.SC_BAD_REQUEST;
+			throw new RuntimeException(statusResponse.toString());
+		}
+
+		String authUrl = credentials.get(AUTH_URL);
+		String currentTokenEndpoint = v3TokensEndpoint;
+		if (authUrl != null && !authUrl.isEmpty()) {
+			currentTokenEndpoint = authUrl + V3_TOKENS_ENDPOINT_PATH;
+		}
+
+		Response response = doPostRequest(currentTokenEndpoint, json);
+		Token token = getTokenFromJson(response);
+
+		return token;
+	}
+
+	@Override
+	public Token reIssueToken(Token token) {
+		return token;
+	}
+
+	@Override
+	public Token getToken(String accessId) {
+		// TODO Implement this method?
+		return null;
+	}
+
+	@Override
+	public boolean isValid(String accessId) {
+		// TODO Implement this method?
+		return true;
+	}
+
+	@Override
+	public Credential[] getCredentials() {
+		return new Credential[] { new Credential(PROJECT_ID, true, null), new Credential(PASSWORD, true, null),
+				new Credential(USER_ID, true, null), new Credential(AUTH_URL, true, null) };
+	}
+
+	@Override
+	public String getAuthenticationURI() {
+
+		StringBuilder sb = new StringBuilder();
+		sb.append("Keystone uri='");
+		sb.append(keystoneUrl);
+		sb.append("'");
+
+		return sb.toString();
+	}
+
+	@Override
+	public Token getForwardableToken(Token originalToken) {
+		return originalToken;
+	}
+
+	private Response doPostRequest(String endpoint, JSONObject json) {
+		HttpResponse response = null;
+		String responseStr = null;
+		try {
+			HttpPost request = new HttpPost(endpoint);
+			request.addHeader(HttpRequestUtil.CONTENT_TYPE, HttpRequestUtil.JSON_CONTENT_TYPE);
+			request.addHeader(HttpRequestUtil.ACCEPT, HttpRequestUtil.JSON_CONTENT_TYPE);
+			request.setEntity(new StringEntity(json.toString(), StandardCharsets.UTF_8));
+			response = getClient().execute(request);
+			responseStr = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
+		} catch (UnknownHostException e) {
+			LOGGER.error("Could not do post request, unknown host.", e);
+			Integer statusResponse = HttpStatus.SC_BAD_REQUEST;
+			throw new RuntimeException(statusResponse.toString());
+		} catch (Exception e) {
+			LOGGER.error("Could not do post request.", e);
+			Integer statusResponse = HttpStatus.SC_BAD_REQUEST;
+			throw new RuntimeException(statusResponse.toString());
+		} finally {
+			try {
+				EntityUtils.consume(response.getEntity());
+			} catch (Throwable t) {
```

Comment:
Why do we catch a Throwable? Wouldn't be better to catch a more specific exception? Or even avoid this exception in finally body.